src/tutorials/pet-shop.md

```diff
@@ -374,6 +376,100 @@ Since arrays can only return a single value given a single key, we create our ow
+
+## Testing the smart contract using Javascript 
+
+<details>
+<summary> Expand This Section </summary>
+Truffle is very flexible when it comes to smart contract testing, in that tests can be written either in JavaScript or Solidity. In this tutorial, we'll be writing our tests in Javascript using chai and mocha library.<br/>
+
+1. Create a new file named `testAdoption.test.js` in the `test/` directory.<br/>
+2. Add the following content to the `testAdoption.test.js` file:<br/>
+
+  ```
+  const Adoption = require("../contracts/Adoption.sol");
+  
+  contract("Adoption", (accounts) => {
+    let adoption;
+    let expectedPetId;
+    
+    before(async() => {
+        adoption = await Adoption.deployed();
+    });
+
+    describe("adopting a pet and retrieving account addresses", async() => {
+      before("adopt a pet using accounts[0]", async () => {
+      await adoption.adopt(8, { from: accounts[0] });
+      expectedAdopter = accounts[0];
+    });
+    });
+  });
+
+  ```
+  We start the contract by importing : 
+  * ```Adoption.sol```:The smart contract we want to test
+  * ```chai```: Gives us various assertions to use in our tests to be used for equality, inequality or emptiness to return a pass/fail from our test.
+
+  **Note**: When writing the contract, we make use of the callback function ```accounts``` that provides us with the accounts in the development or the test network used.
+
+  Then, we make use of async functions : 
+  * To deploy the contract as soon as the test begins for the contract 
+  * To use the asynchronous feature to get the address of the contract and expectedAdopter
+
+  ### Testing the adopt() function 
+
+  To test the ```adopt()``` function, recall that upon success it returns the given ```petId```. We can ensure an ID was returned and that it's correct by comparing the return value of ```adopt()``` to the ID we passed in.
+
+  1. Add the following function within the ```testAdoption.test.js``` test file, after the declaration of ```describe```. 
+
+  ```
+  describe("adopting a pet and retrieving account addresses", async () => {
+    before("adopt a pet using accounts[0]", async () => {
+      await adoption.adopt(8, { from: accounts[0] });
+      expectedAdopter = accounts[0];
+    });
+
+    it("can fetch the address of an owner by pet id", async () => {
+      const adopter = await adoption.adopters(8);
+      assert.equal(adopter, expectedAdopter, "The owner of the adopted pet should be the first account.");
+    });
+  });
+  ```
+
+  Things to notice:
+
+  * We call the smart contract we declared earlier with the ID of expectedPetId.
+  * When called the function within the smart contract, we send metadata for the solidity contract to know that ```msg.sender``` is ```accounts[0]```.
+  * Finally, we pass the actual value, the expected value and a failure message (which gets printed to the console if the test does not pass) to assert.equal().
+
+  ### Testing retrieval of a single pet's owner
```

Comment:
Maybe this section should just be a part of the previous section. Then it would have to ben named something like "Testing the `adopt` function and retrieving adopters' addresses". What do you think?

Reply:
Yes, it is the same section as the previous one so I have removed this section.